### PR TITLE
[RW-12102][risk=low]  [P1] Clicking "view" in the Jupyter Notebook menu automatically downloads files that aren't notebooks

### PR DESCRIPTION
**Pull request by NehaBroad**
For all files with extension **.ipynb** open in iframe
for all other files show the download pop message

https://github.com/all-of-us/workbench/assets/34481816/0becb2af-f69e-410c-b543-9455b068bd01

How to Test:

- One easy way is to copy the file aou-file-tree-policy-extension.js to a public bucket using broad account (i created a new one [aou-test-jupyter-view](https://console.cloud.google.com/storage/browser/aou-test-jupyter-view;tab=objects?forceOnBucketsSortingFiltering=true&project=terra-dev-14905480)) 
- Update the path at https://github.com/all-of-us/workbench/blob/ff3fdc9edd4c205abdd85e8245281327d8de10bc/api/src/main/java/org/pmiops/workbench/leonardo/LeonardoApiClientImpl.java#L161  to "https://storage.googleapis.com/aou-test-jupyter-view/aou-file-tree-policy-extension.js")
- Restart Api server
- Restart jupyter env

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [x] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [x] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

**Review by evrii — APPROVED** [review]
Tested locally and worked as expected.

Thanks for doing this.

File: api/src/main/webapp/static/aou-file-tree-policy-extension.js [hunk 1]
```diff
@@ -31,6 +34,35 @@ define([
...
+          isJupyterFile = false;
+          return false;
+        }
+       });
+      // If all selected Files are jupyter files return now
+      if (!!isJupyterFile) {
+        return true;
+      }
+
+      const affirm = prompt('The All of Us Data Use Policies prohibit you from removing participant-level data from ' +
```
Comment (jmthibault79) [inline api/src/main/webapp/static/aou-file-tree-policy-extension.js:58]: this works but I see that this duplicates the Download button behavior.  Can we make this common code?

File: api/src/main/webapp/static/aou-file-tree-policy-extension.js [hunk 2]
```diff
@@ -6,30 +6,62 @@ define([
...
     // jQuery click handler. We have no way of ordering our click handler ahead
     // of it without digging into jQuery internals. Instead, we insert a span
     // on top of the button to gain click priority.
     $('.dynamic-buttons .download-button').empty().append(
         '<span id="aou-download-button-overlay">Download</span>');
+    $('.dynamic-buttons .view-button').empty().append(
+        '<span id="aou-view-button-overlay">View</span>');
+
+
+    function downloadPolicyPopUp() {
```
Comment (NehaBroad) [inline api/src/main/webapp/static/aou-file-tree-policy-extension.js:27]: i am not sure if this is the right spot for this function, open for suggestion

File: api/src/main/webapp/static/aou-file-tree-policy-extension.js [hunk 3]
```diff
@@ -42,7 +42,7 @@ define([
 
     const viewableFileExtenstions = ['ipynb', 'txt', 'sh'];
     // Clicking the view button will
-    // 1) Open All Jupyter files in a new tab.
+    // 1) Open All Jupyter, text or shell files in a new tab.
```
Comment (jmthibault79) [inline api/src/main/webapp/static/aou-file-tree-policy-extension.js:45]: `isJupyterFile` is no longer a correct name, then

File: api/src/main/webapp/static/aou-file-tree-policy-extension.js (hunk 2 above)
Comment (jmthibault79) [reply api/src/main/webapp/static/aou-file-tree-policy-extension.js:27]: LGTM

**Review by jmthibault79 — APPROVED** [review]

File: api/src/main/webapp/static/aou-file-tree-policy-extension.js (hunk 3 above)
Comment (NehaBroad) [reply api/src/main/webapp/static/aou-file-tree-policy-extension.js:45]: Changed to openInANewTab

**Pull request merged**